test/variable_env_test.py

```diff
@@ -1,36 +1,109 @@
     # seed_test(generated_agents_parallel_v0.env)
-    # seed_test(generated_agents_parallel_v0.env)
 
 
 def test_parallel_generated_agents_conversions():
     parallel_api_test(aec_to_parallel(generated_agents_parallel_v0.env()))
     api_test(parallel_to_aec(generated_agents_parallel_v0.parallel_env()))
 
+    # double-wrapping should produce the same behavior as unwrapped env
+    env1 = generated_agents_parallel_v0.parallel_env()
+    env2 = aec_to_parallel(parallel_to_aec(generated_agents_parallel_v0.parallel_env()))
+
+    # fails (due to parallel_to_aec conversion issues with this environment)
+    # check_environment_deterministic_parallel(env1, env2, 500)
+
+    # double-wrapping should produce deterministic behavior
+    parallel_seed_test(
+        lambda: aec_to_parallel(
+            parallel_to_aec(generated_agents_parallel_v0.parallel_env())
+        )
+    )
+
+    # double- and quadruple- wrapping should cancel out in the same way
+    env1 = aec_to_parallel(parallel_to_aec(generated_agents_parallel_v0.parallel_env()))
+    env2 = aec_to_parallel(
+        parallel_to_aec(
+            aec_to_parallel(
+                parallel_to_aec(generated_agents_parallel_v0.parallel_env())
+            )
+        )
+    )
+    assert type(env1) == type(
+        env2
+    ), "parallel_to_aec and aec_to_parallel wrappers should cancel out. Result: {type(env1)}, {type(env2)}."
+
+    # ensure double- and quadruple- wrapped environments perform the same
+    check_environment_deterministic_parallel(env1, env2, 500)
+
+    # quadruple-wrapping should produce deterministic behavior
+    parallel_seed_test(
+        lambda: aec_to_parallel(
+            parallel_to_aec(
+                aec_to_parallel(
+                    parallel_to_aec(generated_agents_parallel_v0.parallel_env())
+                )
+            )
+        )
+    )
+
+
+@pytest.mark.skip(
+    reason="parallel_to_aec wrapper produces nondeterministic behavior with this environment."
+)
+def test_parallel_generated_agents_conversions_aec():
+    # fails
+    seed_test(
+        lambda: parallel_to_aec(
+            aec_to_parallel(
+                parallel_to_aec(generated_agents_parallel_v0.parallel_env())
+            )
+        )
+    )
+
+    # single- and triple-wrapping should cancel out in the same way
     env1 = parallel_to_aec(generated_agents_parallel_v0.parallel_env())
     env2 = parallel_to_aec(
         aec_to_parallel(parallel_to_aec(generated_agents_parallel_v0.parallel_env()))
     )
+    assert type(env1) == type(
+        env2
+    ), "parallel_to_aec and aec_to_parallel wrappers should cancel out. Result: {type(env1)}, {type(env2)}."
+
+    # fails because using parallel_to_aec with this environment can produce nondeterministic behavior
     check_environment_deterministic(env1, env2, 500)
 
 
 if __name__ == "__main__":
     test_generated_agents_aec()
     test_generated_agents_parallel()
+    test_parallel_generated_agents_conversions()
```

Comment:
This should just be removed. The file should be run with `pytest -v -rw test/variable_env_test.py` instead.